libraries/networking/src/udt/CongestionControl.cpp

```diff
@@ -0,0 +1,205 @@
+//
+//  CongestionControl.cpp
+//  libraries/networking/src/udt
+//
+//  Created by Clement on 7/23/15.
+//  Copyright 2015 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#include "CongestionControl.h"
+#include "Packet.h"
+
+using namespace udt;
+using namespace std::chrono;
+
+static const double USECS_PER_SECOND = 1000000.0;
+
+void CongestionControl::setPacketSendPeriod(double newSendPeriod) {
+    Q_ASSERT_X(newSendPeriod >= 0, "CongestionControl::setPacketPeriod", "Can not set a negative packet send period");
+    
+    if (_maxBandwidth > 0) {
+        // anytime the packet send period is about to be increased, make sure it stays below the minimum period,
+        // calculated based on the maximum desired bandwidth
+        double minPacketSendPeriod = USECS_PER_SECOND / (((double) _maxBandwidth) / _mss);
+        _packetSendPeriod = std::max(newSendPeriod, minPacketSendPeriod);
+    } else {
+        _packetSendPeriod = newSendPeriod;
+    }
+}
+
+void DefaultCC::init() {
```

Comment:
Good question - we were basically just following what UDT was doing, but I agree. Will move to ctor